src/blockMonitor/blockProcessor.ts

```diff
@@ -211,7 +227,19 @@ export class BlockProcessor<TBlock extends IBlockStub> extends StartStopService
                 // starting from observedBlock, add to the cache and download the parent, until the return value signals that block is attached
                 let curBlock = observedBlock;
                 while (true) {
-                    const addResult = await this.mBlockCache.addBlock(curBlock)
+                    let addResult: BlockAddResult | null = null;
+
+                    try {
+                        await this.blockItemStoreLock.acquire();
+                        await this.blockItemStore.withBatch(async () => {
+                            // We execute all the writes in the components as a consequence of updating this block
```

Comment:
I think you need to be more specific in your comment here - and explain a bit more what could go wrong if you didnt use the batch.

Reply:
Comment expanded in 5828043.